### PR TITLE
feat: Display Upcoming earnings of current week instead of my total earnings - MEED-1134 - Meeds-io/MIPs#10

### PR DESCRIPTION
**Pull request by MayTekayaa**
This change will display the Upcoming earnings of current week instead of my total earnings.

### CODE REVIEW

File: wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java [hunk 1]
```diff
@@ -206,6 +206,17 @@ public RewardReport computeRewards(LocalDate date) {
     return rewardReport;
   }
 
+  @Override
+  public RewardReport computeRewardsByUser(LocalDate date, Long technicalUserId) {
+
```
Comment (AzmiTouil) [inline wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java]:
```suggestion
```

File: wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java [hunk 2]
```diff
@@ -206,6 +206,17 @@ public RewardReport computeRewards(LocalDate date) {
...
+    RewardReport rewardReport = computeRewards(date);
+
```
Comment (AzmiTouil) [inline wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java]:
```suggestion
```

File: wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java [hunk 3]
```diff
@@ -206,6 +206,17 @@ public RewardReport computeRewards(LocalDate date) {
...
+    Set<WalletReward> wallet = rewardReport.getRewards().stream().filter(reward -> reward.getIdentityId() == technicalUserId).collect(Collectors.toSet());
+    rewardReport.setRewards(wallet);
```
Comment (AzmiTouil) [inline wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java]:
```suggestion
    Set<WalletReward> rewards = rewardReport.getRewards().stream().filter(reward -> reward.getIdentityId() == technicalUserId).collect(Collectors.toSet());
    rewardReport.setRewards(rewards);
```

File: wallet-api/src/main/java/org/exoplatform/wallet/reward/service/RewardReportService.java [hunk 4]
```diff
@@ -38,6 +38,18 @@ public interface RewardReportService {
...
+   * Compute rewards By User
+   *
+   * @param localDate a {@link LocalDate} inside the period time
+   *          that will be retrieved
+   * @param technicalUserId a {@link Long} user identity technical id
+   *          that will be retrieved
+   * @return a {@link Set} of {@link WalletReward} with the details of sent
+   *         tokens and tokens to send
+   */
+  RewardReport computeRewardsByUser(LocalDate localDate, Long technicalUserId);
```
Comment (AzmiTouil) [inline wallet-api/src/main/java/org/exoplatform/wallet/reward/service/RewardReportService.java]:
```suggestion
  RewardReport computeRewardsByUser(LocalDate localDate, long userIdentityId);
```

File: wallet-api/src/main/java/org/exoplatform/wallet/reward/service/RewardReportService.java [hunk 5]
```diff
@@ -38,6 +38,18 @@ public interface RewardReportService {
    */
   RewardReport computeRewards(LocalDate localDate);
 
+  /**
+   * Compute rewards By User
+   *
+   * @param localDate a {@link LocalDate} inside the period time
+   *          that will be retrieved
+   * @param technicalUserId a {@link Long} user identity technical id
+   *          that will be retrieved
```
Comment (AzmiTouil) [inline wallet-api/src/main/java/org/exoplatform/wallet/reward/service/RewardReportService.java]:
```suggestion
   * @param userIdentityId User technical identifier ({@link Identity#getId()})
```

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

File: wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java (hunk 1 above)
Comment (MayTekayaa) [reply wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java]: done

File: wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java (hunk 2 above)
Comment (MayTekayaa) [reply wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java]: done

File: wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java (hunk 3 above)
Comment (MayTekayaa) [reply wallet-reward-services/src/main/java/org/exoplatform/wallet/reward/service/WalletRewardReportService.java]: done

File: wallet-api/src/main/java/org/exoplatform/wallet/reward/service/RewardReportService.java (hunk 4 above)
Comment (MayTekayaa) [reply wallet-api/src/main/java/org/exoplatform/wallet/reward/service/RewardReportService.java]: done

File: wallet-api/src/main/java/org/exoplatform/wallet/reward/service/RewardReportService.java (hunk 5 above)
Comment (MayTekayaa) [reply wallet-api/src/main/java/org/exoplatform/wallet/reward/service/RewardReportService.java]: done

**Review by AzmiTouil — APPROVED** [review]

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**